process-cmds.sh

```diff
@@ -94,6 +94,166 @@ process-iperf() {
+
+  # Start tcpdump
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp\""
+  kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp" > "${TCPDUMP_FILENAME}"
+  cat "${TCPDUMP_FILENAME}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Record ethtool stats
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'\""
+  ethtoolend=$(kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'")
+  echo "${ethtoolend}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  rxpktend=$(echo "$ethtoolend" | sed -n "s/^rx_packets:\s\+//p" | sed "s/[^0-9]//g")
+  txpktend=$(echo "$ethtoolend" | sed -n "s/^tx_packets:\s\+//p" | sed "s/[^0-9]//g")
+
+  rxcount=$(( rxpktend - rxpktstart ))
+  txcount=$(( txpktend - txpktstart ))
+
+  echo "Ethtool results for ${TEST_CLIENT_CLIENT_VF_REP}:"
+  echo "================================================="
+  echo "RX Packets: ${rxpktend} - ${rxpktstart} = ${rxcount}"
+  echo "TX Packets: ${txpktend} - ${txpktstart} = ${txcount}"
+
+  # Wait for Iperf to finish
+  wait $IPERF_PID
+
+  # Concatenate the background Iperf results into the same file
+  cat "${IPERF_FILENAME}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Dump command output
+  if [ "$VERBOSE" == true ]; then
+    echo "Full Tcpdump Output (from ${HWOL_VALIDATION_FILENAME}):"
+    cat ${TCPDUMP_FILENAME}
+    echo "Full Iperf Output (from ${HWOL_VALIDATION_FILENAME}):"
+    cat ${IPERF_FILENAME}
+  else
+    echo "Summary Tcpdump Output (see ${HWOL_VALIDATION_FILENAME} for full detail):"
+    tail ${TCPDUMP_FILENAME}
+    echo "Summary Iperf Output (see ${HWOL_VALIDATION_FILENAME} for full detail):"
+    cat ${IPERF_FILENAME} | grep -B 1 -A 1 "sender"
+  fi
+
+  cat ${IPERF_FILENAME} | grep -cq "sender" && retVal=0 || retVal=1
+
+  if (( rxcount > THRESHOLD_PKT_COUNT )) || (( txcount > THRESHOLD_PKT_COUNT )); then
+    retVal=1
+  fi
+
+  # Cleanup temporary files
+  rm "${IPERF_FILENAME}"
+  rm "${TCPDUMP_FILENAME}"
+
+  return $retVal
+}
+
+process-hw-offload-validation() {
+  # The following VARIABLES are used by this function:
+  #     TEST_CLIENT_NODE
+  #     TEST_SERVER_NODE
+  #     TEST_CLIENT_POD
+  #     TEST_FILENAME
+  #     TEST_SERVER_IPERF_DST
+  #     TEST_SERVER_IPERF_DST_PORT
+
+  HWOL_VALIDATION_FILENAME="${HW_OFFLOAD_LOGS_DIR}/${TEST_FILENAME}"
+
+  echo "kubectl get pods -n ${FT_NAMESPACE} --selector=name=${TOOLS_POD_NAME} -o wide | grep -w \"${TEST_CLIENT_NODE}\" | awk -F' ' '{print $1}'"
+  TOOLS_CLIENT_POD=$(kubectl get pods -n ${FT_NAMESPACE} --selector=name=${TOOLS_POD_NAME} -o wide | grep -w "${TEST_CLIENT_NODE}" | awk -F' ' '{print $1}')
```

Comment:
I use a new pattern in [variables.sh-L609](https://github.com/Billy99/ovn-kuber-traffic-flow-tests/blob/main/variables.sh#L609) where I try to limit the kubectl calls to speed it up some. I haven't gone back and fixed other places, but try to follow it in new code. So I pipe the base kubectl command into a TMP_VAR, the run multiple "echo TMP_VAR | grep ... | awk ...". Example here:
```suggestion
  TMP_OUTPUT=$(kubectl get pods -n ${FT_NAMESPACE} --selector=name=${TOOLS_POD_NAME} -o wide)
  TOOLS_CLIENT_POD=$(echo "${TMP_OUTPUT}" | grep -w "${TEST_CLIENT_NODE}" | awk -F' ' '{print $1}')
  TOOLS_SERVER_POD=$(echo "${TMP_OUTPUT}" | grep -w "${TEST_SERVER_NODE}" | awk -F' ' '{print $1}')  
```